query-engine/js-connectors/src/proxy.rs

```diff
@@ -335,274 +467,568 @@ impl Proxy {
     }
 
     pub fn is_healthy(&self) -> napi::Result<bool> {
-        // TODO: call `is_healthy` in a blocking fashion, returning its result as a boolean.
-        unimplemented!();
+        let result_arc = Arc::new((Mutex::new(None), Condvar::new()));
+        let result_arc_clone: Arc<(Mutex<Option<bool>>, Condvar)> = result_arc.clone();
+
+        let set_value_callback = move |value: bool| {
+            let (lock, cvar) = &*result_arc_clone;
+            let mut result_guard = lock.lock().unwrap();
+            *result_guard = Some(value);
+            cvar.notify_one();
+
+            Ok(())
+        };
+
+        // Should anyone find a less mind-boggling way to retrieve the result of a synchronous JS
+        // function, please do so.
+        self.is_healthy.call_with_return_value(
+            (),
+            napi::threadsafe_function::ThreadsafeFunctionCallMode::Blocking,
+            set_value_callback,
+        );
```

Comment:
So @jkomyno also raised a very valid question, which is what is the current use case for is_healthy in quaint? and could it be removed from the `Queryable` interface? This would be a brute-force solution, to the above complex code that sits waiting for the operation to complete.


Said code is pretty hacky, but Alberto explored writing it in a different way and didn't find something better. The cause is that the js code for this function is synchronous, but we cannot wrap it successfully in the proxy under a simple `JsFunction`.

Maybe @Brooooooklyn knows a better way around this?

Reply:
@miguelff does this function always call in the Node.js JavaScript thread? If that's the case, maybe `Ref<JsFunction>` works

Reply:
Hey @Brooooooklyn, unfortunately declaring`is_healthy: Ref<JsFunction>` doesn't seem to work, as:
- that's not a `NapiValue`
- it's non `Clone`able

```
error[E0277]: the trait bound `napi::Ref<JsFunction>: NapiValue` is not satisfied
   --> query-engine/js-connectors/src/proxy.rs:56:35
    |
56  |     let is_healthy = js_connector.get_named_property("isHealthy")?;
    |                                   ^^^^^^^^^^^^^^^^^^ the trait `NapiValue` is not implemented for `napi::Ref<JsFunction>`
    |
```